tests/org.eclipse.ui.workbench.texteditor.tests/src/org/eclipse/ui/workbench/texteditor/tests/IFindReplaceUIAccess.java

```diff
@@ -0,0 +1,59 @@
+package org.eclipse.ui.workbench.texteditor.tests;
+
+import java.util.Set;
+
+import org.eclipse.swt.widgets.Widget;
+
+import org.eclipse.jface.text.IFindReplaceTarget;
+
+import org.eclipse.ui.internal.findandreplace.IFindReplaceLogic;
+import org.eclipse.ui.internal.findandreplace.SearchOptions;
+
+
+/**
+ * Wraps UI-Access for different Find/Replace-UIs
+ */
+interface IFindReplaceUIAccess {
+
+	IFindReplaceTarget getTarget();
+
+	void close();
+
+	void ensureHasFocusOnGTK();
+
+	void unselect(SearchOptions option);
+
+	void select(SearchOptions option);
+
+	void simulateEnterInFindInputField(boolean shiftPressed);
+
+	void simulateKeyPressInFindInputField(int keyCode, boolean shiftPressed);
+
+	String getFindText();
+
+	String getReplaceText();
+
+	void setFindText(String text, boolean grabFocus);
```

Comment:
Would be better to remove the `grabFocus` parameter and set focus on the input field separately. It is only used at a single place, at which the find combo can be accessed directly (via `dialog.getFindCombo()`) to set the focus.

Reply:
Done in https://github.com/eclipse-platform/eclipse.platform.ui/commit/49851d64fd5a67634b90f951a0aeddf3da74bfaa